apps/etterlatte-vedtaksvurdering/src/main/kotlin/vedtaksvurdering/VedtaksvurderingService.kt

```diff
@@ -78,52 +80,82 @@ class VedtaksvurderingService(
         return repository.hentVedtak(behandlingId)
     }
 
-    suspend fun populerOgHentFellesVedtak(behandlingId: UUID, accessToken: String): Vedtak? {
-        val vedtak = hentFellesVedtakMedUtbetalingsperioder(behandlingId)
-        if (vedtak?.vedtakFattet != null) {
-            return vedtak
+    suspend fun opprettEllerOppdaterVedtak(behandlingId: UUID, accessToken: String): Vedtak {
+        val vedtak = hentVedtak(behandlingId)
+        if (vedtak == null) {
+            val vedtakData = hentDataForVedtak(behandlingId, accessToken)
+
+            val virk = vedtakData.second.virkningstidspunkt.dato.atDay(1)
+            repository.opprettVedtak(
+                behandlingId,
+                vedtakData.third.sak,
```

Comment:
Bruk av `Triple` gjør det veldig kronglete å forstå hva dette faktisk er. Om du absolutt må bruke `Pair`, `Triple`, e.l. burde du se litt på [destrukturering i Kotlin](https://kotlinlang.org/docs/destructuring-declarations.html). Alternativt opprett ny klasse for å holde på dataene.

Reply:
267de1e8fc532813769a4e014381721d7ed1a366